src/main/scala/za/co/absa/spark/commons/implicits/StructTypeImplicits.scala

```diff
@@ -0,0 +1,432 @@
+              arrayType.elementType match {
+                case _: StructType => true
+                case _ => false
+              }
+            case _ => false
+          }
+        case None => false
+      }
+    }
+
+    /**
+     * Get nullability of a field from a text path and a given schema
+     *
+     * @param path   The dot-separated path to the field
+     * @return Some(nullable) or None if the field does not exist
+     */
+    def getFieldNullability(path: String): Option[Boolean] = {
+      getField(path).map(_.nullable)
+    }
+
+    /**
+     * Checks if a field specified by a path and a schema exists
+     * @param path   The dot-separated path to the field
+     * @return       True if the field exists false otherwise
+     */
+    def fieldExists(path: String): Boolean = {
+      getField(path).nonEmpty
+    }
+    /**
+     * Returns all renames in the provided schema.
+     * @param includeIfPredecessorChanged  if set to true, fields are included even if their name have not changed but
+     *                                     a predecessor's (parent, grandparent etc.) has
+     * @return        the keys of the returned map are the columns' names after renames, the values are the source columns;
+     *                the name are full paths denoted with dot notation
+     */
+    def getRenamesInSchema(includeIfPredecessorChanged: Boolean = true): Map[String, String] = {
+
+      def getRenamesRecursively(path: String,
+                                sourcePath: String,
+                                struct: StructType,
+                                renamesAcc: Map[String, String],
+                                predecessorChanged: Boolean): Map[String, String] = {
+        import za.co.absa.spark.commons.implicits.StructFieldImplicits.StructFieldEnhancements
+
+        struct.fields.foldLeft(renamesAcc) { (renamesSoFar, field) =>
+          val fieldFullName = appendPath(path, field.name)
+          val fieldSourceName = field.getMetadataString(MetadataKeys.SourceColumn).getOrElse(field.name)
+          val fieldFullSourceName = appendPath(sourcePath, fieldSourceName)
+
+          val (renames, renameOnPath) = if ((fieldSourceName != field.name) || (predecessorChanged && includeIfPredecessorChanged)) {
+            (renamesSoFar + (fieldFullName -> fieldFullSourceName), true)
+          } else {
+            (renamesSoFar, predecessorChanged)
+          }
+
+          field.dataType match {
+            case st: StructType => getRenamesRecursively(fieldFullName, fieldFullSourceName, st, renames, renameOnPath)
+            case at: ArrayType  => getStructInArray(at.elementType).fold(renames) { item =>
+              getRenamesRecursively(fieldFullName, fieldFullSourceName, item, renames, renameOnPath)
+            }
+            case _              => renames
+          }
+        }
+      }
+
+      @tailrec
+      def getStructInArray(dataType: DataType): Option[StructType] = {
```

Comment:
Very similar to `goThroughArrayDataType`. Maybe that should be one private method outside of these methods?